### ISSUE TITLE
No longer scrobbling in 1.5.7 (Android 9)

### ISSUE THREAD

**Issue by Thorarin** [tgwizard/sls#459]
I'm using Pulsar as a music player. My last scrobble was on February 4th 2019. Pulsar hasn't updated for a few weeks, so I suspect it has something to do with the update that was released to te store on February 3rd (version reads 1.5.7).

Pulsar is detected and scrobbling for it is enabled. 
Android 9 on a Samsung S9+.

Copy to clipboard gives "no current song available" and the local cache stays empty. I can't find any diagnostic information to share with you.

**Comment by nguyenmq** [comment]
I just wanted to +1 this issue because I'm experiencing the same problem on my S9 after upgrading to Android 9. I also use Pulsar as my music player.

Additionally, after reinstalling Simple Scrobbler, there are no music apps detected.

**Comment by a93-39a** [comment]
##### Preferred Feature format
  Request for Feature:
  Reason:
##### Preferred issue forma
  * Phone Model:
  * Android Version:
  * SLS Version: (5.3)
  * Music App(s) In Use: (ex. YT Music, Spotify)
  * Detected Music Apps(s): (List what apps SLS detected)
  * Enabled Options: (List what options you have enabled)
  * Issue: (Description of What The Issue Is)
  * Parameters: (ex. When, Where, Why & How The Issue Occurs)

**Comment by a93-39a** [comment]
I am using Android 9 right now.
It could be an issue with your options.

**Comment by Thorarin** [comment]
It happens with all the default options. 

I deleted all app data, logged back into Last.fm. Then I play something in both Pulsar and Google Play Music, but the list of detected apps stays empty, like @nguyenmq described.

**Comment by a93-39a** [comment]
It has to do with Power Optimization being enabled.
Please try enabling "OnGoing" and "Phone notifications" in your Options please.

**Comment by Thorarin** [comment]
I've just enabled both options, but there is no change in behavior. Still no apps detected.

**Comment by a93-39a** [comment]
Well I am going to post a merge soon.
The README.md will have a discord chat room we can talk.
Also there is a Gitter chat room on the README.md
I cannot diagnose remotely, all we have are words to discuss the issue.
This is very strange behavior. I will do my best to look into it.

**Comment by a93-39a** [comment]
@Thorarin 
Is there a change this is what happened [https://github.com/tgwizard/sls/issues/451](https://github.com/tgwizard/sls/issues/451) ??

**Comment by Thorarin** [comment]
@a93-39a sounds like it might be the same problem.

Perhaps I can find some time to do some debugging, but my experience with Android development is very limited, so I can't make any promises.

**Comment by a93-39a** [comment]
No. That's fine, I can probably review the code and discern the issue.

**Comment by a93-39a** [comment]
Seems to be some kind of SQLite issue.
This is a much older piece of code if I'm not mistaken.

I can send a trial version and it will be in .apk (and not a signed version) format so you would need to uninstall the current version.
I can send you a copy of a potential fix if you like.
Sounds good?

**Comment by Thorarin** [comment]
Sure, there should be contact details in my profile if needed.

**Comment by a93-39a** [comment]
[We have a missing close database here](https://github.com/tgwizard/sls/blob/master/app/src/main/java/com/adam/aslfms/receiver/MusicAPI.java#L197-L202)
Can cause database corruption.
Most likely you should be able to clear the cache and the app should work again.
